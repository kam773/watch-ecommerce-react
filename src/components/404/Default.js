import React, { Component } from "react";
import { Container, Row, Col } from "reactstrap";

export default class Default extends Component {
  render() {
    return (
      <Container>
        <Row className="my-5">
          <Col
            xs="10"
            className="mx-auto text-center text-title text-uppercase my-5"
          >
            <h1 className="display-3 my-5">404</h1>
            <h1>error</h1>
            <h2>page not found</h2>
            <h3>
              the requested URL{" "}
              <span className="text-danger">
                {this.props.location.pathname}
              </span>{" "}
              was not found
            </h3>
          </Col>
        </Row>
      </Container>
    );
  }
}
