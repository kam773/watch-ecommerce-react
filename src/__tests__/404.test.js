import React from 'react';
import Default from '../components/404/Default';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

describe("404", () => {
    it("renders without crashing", () => {
        const location = { pathname: '/ddd' };
        const tree = renderer.create(<Default location={location}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('has h1 tag with class display-3', () => {
        const location = { pathname: '/ddd' };
        const wrapper = shallow(<Default location={location}/>)
        expect(wrapper.find('h1')).toHaveLength(2);
    });
    it('has h2 tag', () => {
        const location = { pathname: '/ddd' };
        const wrapper = shallow(<Default location={location}/>)
        expect(wrapper.find('h2')).toHaveLength(1);
    });
    it('has h3', () => {
        const location = { pathname: '/ddd' };
        const wrapper = shallow(<Default location={location}/>)
        expect(wrapper.find('h3')).toHaveLength(1);
    })

})