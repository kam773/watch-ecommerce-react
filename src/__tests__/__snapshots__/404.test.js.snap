// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 renders without crashing 1`] = `
<div
  className="container"
>
  <div
    className="row my-5"
  >
    <div
      className="col-10 mx-auto text-center text-title text-uppercase my-5"
    >
      <h1
        className="display-3 my-5"
      >
        404
      </h1>
      <h1>
        error
      </h1>
      <h2>
        page not found
      </h2>
      <h3>
        the requested URL 
        <span
          className="text-danger"
        >
          /ddd
        </span>
         was not found
      </h3>
    </div>
  </div>
</div>
`;
